src/services/catalog.service.ts

```diff
@@ -96,25 +96,39 @@ export class CatalogService {
     return new TemplateItem(resource, this);
   }
 
-  public getCategoriesBySubCategories(tags: any) {
+  public getCategoriesBySubCategories(itemTags: any) {
     let catsBySubCats = {};
     let otherId = 'other';
-
-    _.each(tags, (tag) => {
-      _.each(this.categories, (category) => {
-        let subCat: any = _.find(category.subCategories, (subCategory: any) => {
-              return subCategory.id === tag || _.includes(subCategory.categoryAliases, tag);
-            }
-        );
-        if (subCat) {
-          catsBySubCats[subCat.id] = category.id;
-          return false;
+    let filteredSubCats: any;
+
+    _.each(this.categories, (category) => {
+      if (category.tags) {
+        if (!_.isEmpty(this.getMatchingTags(category.tags, itemTags))) {
+          filteredSubCats = this.filterSubCatsByTags(category.subCategories, itemTags);
+          if (!_.isEmpty(filteredSubCats)) {
+            _.each(filteredSubCats, (subCat) => {
+              catsBySubCats[subCat.id] = category.id;
+            });
+          } else {
+            catsBySubCats[otherId] = category.id;
```

Comment:
sorry yes, your particular example is fine, thats not the example i'm saying is broken. any single item can be categorized into many different main categories and subcategories.  if you instead had:

```
"mycat-a-other": {
    kind: 'ServiceClass',
    "metadata": {
      "name": "mycat-a-other",
      "uid": "11",
    },
    osbTags: [
      "mycat-a",
      "mycat-b",
      "foobar"
    ]
  }
```

Then what should happen is that it appears both in `My Category A -> Other` AND in `My Category B -> Other`  but what will happen is that it will only show up in the second one.